zcash_client_sqlite/src/error.rs

```diff
@@ -185,12 +184,13 @@ impl fmt::Display for SqliteClientError {
             SqliteClientError::BalanceError(e) => write!(f, "Balance error: {}", e),
             SqliteClientError::NoteFilterInvalid(s) => write!(f, "Could not evaluate filter query: {:?}", s),
             #[cfg(feature = "transparent-inputs")]
-            SqliteClientError::ReachedGapLimit(account_id, bad_index) => write!(f,
-                "The proposal cannot be constructed until transactions with previously reserved ephemeral address outputs have been mined. \
-                 The ephemeral address in account {account_id:?} at index {bad_index} could not be safely reserved.",
+            SqliteClientError::ReachedGapLimit(bad_index) => write!(f,
+                "The proposal cannot be constructed until transactions with outputs to previously reserved ephemeral addresses have been mined. \
+                 The ephemeral address at index {bad_index} could not be safely reserved.",
```

Comment:
```suggestion
                "The proposal cannot be constructed until transactions with outputs to previously reserved transparent addresses have been mined. \
                 The transparent address at index {bad_index} could not be safely reserved.",
```
I think this now does not only occur for ephemeral addresses.

Reply:
The caller provided the account index for proposal construction, so we don't need to return it (and we don't have the account UUID in the place that the error is generated, and it would be a hassle to look it up.)

Reply:
`KeyScope` parameter has been added.